app/services/plugin.service.ts

```diff
@@ -142,54 +144,56 @@ export class PluginService {
         this.runInputCode = sandboxService.compileScript('sinap.__program.then((program) => program.run(sinap.__input))');
     }
 
-    public getInterpreter(withGraph: Core.Graph): Promise<Program> {
-        let kind = withGraph.plugin.kind;
-        let context = this.interpreters.get(kind);
-        let graph = new Graph(withGraph);
-        var result: Promise<Program>;
-
-        let processContext = (context: any): Promise<Program> => {
-            context.sinap.__graph = graph;
-            console.log(context);
-            return this.interpretCode.runInContext(context);
-        };
-
-        if (context) {
-            result = processContext(context);
-        } else {
-            result = this.loadPlugin(kind, "./build/plugins/dfa-interpreter.js") // TODO: Put real file in here.
-                .then((cont) => {
-                    context = cont;
-                    this.interpreters.set(kind, context);
-                    return processContext(context);
+    public getInterpreter(graph: Core.Graph): Promise<Program> {
+        // TODO: consider: this could really just be a cast and 
+        // graph.plugin.context
+        return this.getPlugin(graph.plugin.kind).context.then((context) => {
```

Comment:
The issue appears to have been trying to cache plugins inside of a map. We should investigate why this is the case since it would be ideal to avoid file reading and compiling graphs as much as possible.